distributed/utils_test.py

```diff
@@ -864,6 +864,7 @@ async def start_cluster(
         await asyncio.sleep(0.01)
         if time() > start + 30:
             await asyncio.gather(*(w.close(timeout=1) for w in workers))
+            assert not s.events["invalid-worker-transition"]
```

Comment:
I'd be a little concerned about erroring out within this cleanup logic? Maybe pull the events off the scheduler, `await s.close()`, then assert? Same for the other one. Or maybe it doesn't matter, not sure.

Reply:
It would also be good to test (in `test_utils_test.py`) that this works as expected.

Reply:
Good idea.  See recent commits.  I also invested a little in nicely formatted outputs.  Dumping the commit message below:

Now we get errors like the following for InvalidTransition

```
Worker: tcp://127.0.0.1:36443
InvalidTransition: inc-03d935909bba38f9a49655e867cbf56a :: released->foo
  Story:
    ('inc-03d935909bba38f9a49655e867cbf56a', 'compute-task', 'compute-task-1650136473.0490825', 1650136473.049759)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'released', 'waiting', 'waiting', {'inc-03d935909bba38f9a49655e867cbf56a': 'ready'}, 'compute-task-1650136473.0490825', 1650136473.0498002)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'waiting', 'ready', 'ready', {'inc-03d935909bba38f9a49655e867cbf56a': 'executing'}, 'compute-task-1650136473.0490825', 1650136473.0498285)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'ready', 'executing', 'executing', {}, 'compute-task-1650136473.0490825', 1650136473.0498428)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'put-in-memory', 'compute-task-1650136473.0490825', 1650136473.0505066)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'executing', 'memory', 'memory', {}, 'compute-task-1650136473.0490825', 1650136473.0505476)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'ensure-task-exists', 'memory', 'compute-task-1650136473.0510707', 1650136473.0558414)
    ('free-keys', ('inc-03d935909bba38f9a49655e867cbf56a',), 'memory-released-1650136473.05692', 1650136473.0609956)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'release-key', 'memory-released-1650136473.05692', 1650136473.0610087)
    ('inc-03d935909bba38f9a49655e867cbf56a', 'memory', 'released', 'released', {}, 'memory-released-1650136473.05692', 1650136473.061037)
```

We also test that this shows up in gen_cluster tests

Reply:
> I'd be a little concerned about erroring out within this cleanup logic? Maybe pull the events off the scheduler, await s.close(), then assert? Same for the other one. Or maybe it doesn't matter, not sure.

Did this too.  Good catch.